lib/utils/meta_manager.py

```diff
@@ -0,0 +1,45 @@
+from utils.re_utils import execute_query
+
+SAMPLE_NODE_COLLECTION = "samples_nodes"
+SAMPLE_SAMPLE_COLLECTION = "samples_sample"
+
+META_AQL_TEMPLATE = f"""
+        let version_ids = (for sample_id in @sample_ids
+            let doc = DOCUMENT({SAMPLE_SAMPLE_COLLECTION}, sample_id.id)
+            RETURN {{
+                'id': doc.id,
+                'version_id': doc.vers[sample_id.version - 1],
+                'version': sample_id.version
+            }}
+        )
+        let node_metas = (for version_id in version_ids
+            for node in {SAMPLE_NODE_COLLECTION}
+                FILTER node.id == version_id.id AND node.uuidver == version_id.version_id
+                LIMIT @num_sample_ids
```

Comment:
Unclear on why this line is needed, are there duplicates that occur after the correct nodes? Seems a little strange. It seems like if a sample has multiple nodes that this will break (i dont beleive any do at the moment, but).

Reply:
Re: a slack conversation. This was taken from the sample set filtering code. I have opened a ticket to investigate this further in the future, but it should work for now. https://kbase-jira.atlassian.net/browse/SAM-143